libc/cmake/modules/LLVMLibCTestRules.cmake

```diff
@@ -15,48 +15,60 @@
 function(get_object_files_for_test result skipped_entrypoints_list)
   set(object_files "")
   set(skipped_list "")
-  foreach(dep IN LISTS ARGN)
-    if (NOT TARGET ${dep})
-      # Skip any tests whose dependencies have not been defined.
-      list(APPEND skipped_list ${dep})
-      continue()
-    endif()
-    get_target_property(dep_type ${dep} "TARGET_TYPE")
-    if(NOT dep_type)
-      # Target for which TARGET_TYPE property is not set do not
-      # provide any object files.
-      continue()
-    endif()
+  set(checked_list "")
+  set(unchecked_list "${ARGN}")
+  list(REMOVE_DUPLICATES unchecked_list)
+  list(LENGTH unchecked_list length)
 
-    if(${dep_type} STREQUAL ${OBJECT_LIBRARY_TARGET_TYPE})
-      get_target_property(dep_object_files ${dep} "OBJECT_FILES")
-      if(dep_object_files)
-        list(APPEND object_files ${dep_object_files})
-      endif()
-    elseif(${dep_type} STREQUAL ${ENTRYPOINT_OBJ_TARGET_TYPE})
-      get_target_property(is_skipped ${dep} "SKIPPED")
-      if(is_skipped)
+  while(length)
+    set(indirect_list "")
+
+    foreach(dep IN LISTS unchecked_list)
+      if (NOT TARGET ${dep})
+        # Skip any tests whose dependencies have not been defined.
         list(APPEND skipped_list ${dep})
         continue()
       endif()
-      get_target_property(object_file_raw ${dep} "OBJECT_FILE_RAW")
-      if(object_file_raw)
-        list(APPEND object_files ${object_file_raw})
+      get_target_property(dep_type ${dep} "TARGET_TYPE")
+      if(NOT dep_type)
+        # Target for which TARGET_TYPE property is not set do not
+        # provide any object files.
+        continue()
       endif()
-    elseif(${dep_type} STREQUAL ${ENTRYPOINT_OBJ_VENDOR_TARGET_TYPE})
-      # We skip tests for all externally implemented entrypoints.
-      list(APPEND skipped_list ${dep})
-      continue()
-    endif()
 
-    get_target_property(indirect_deps ${dep} "DEPS")
-    get_object_files_for_test(
-        indirect_objfiles indirect_skipped_list ${indirect_deps})
-    list(APPEND object_files ${indirect_objfiles})
-    if(indirect_skipped_list)
-      list(APPEND skipped_list ${indirect_skipped_list})
-    endif()
-  endforeach(dep)
+      if(${dep_type} STREQUAL ${OBJECT_LIBRARY_TARGET_TYPE})
+        get_target_property(dep_object_files ${dep} "OBJECT_FILES")
+        if(dep_object_files)
+          list(APPEND object_files ${dep_object_files})
+        endif()
+      elseif(${dep_type} STREQUAL ${ENTRYPOINT_OBJ_TARGET_TYPE})
+        get_target_property(is_skipped ${dep} "SKIPPED")
+        if(is_skipped)
+          list(APPEND skipped_list ${dep})
+          continue()
+        endif()
+        get_target_property(object_file_raw ${dep} "OBJECT_FILE_RAW")
+        if(object_file_raw)
+          list(APPEND object_files ${object_file_raw})
+        endif()
+      elseif(${dep_type} STREQUAL ${ENTRYPOINT_OBJ_VENDOR_TARGET_TYPE})
+        # We skip tests for all externally implemented entrypoints.
```

Comment:
[nit] Skip tests for externally implemented entrypoints.

Reply:
Done.